src/Cli/dotnet/commands/dotnet-sln/remove/Program.cs

```diff
@@ -18,16 +18,27 @@ internal class RemoveProjectFromSolutionCommand : CommandBase
         private readonly string _fileOrDirectory;
         private readonly IReadOnlyCollection<string> _arguments;
 
-        public RemoveProjectFromSolutionCommand(
-            ParseResult parseResult) : base(parseResult)
+        public RemoveProjectFromSolutionCommand(ParseResult parseResult) : base(parseResult)
         {
-            _arguments = (parseResult.ValueForArgument<IEnumerable<string>>(SlnRemoveParser.ProjectPathArgument) ?? Array.Empty<string>()).ToList().AsReadOnly();
+            _fileOrDirectory = parseResult.ValueForArgument<string>(SlnCommandParser.SlnArgument);
+
+            _arguments = parseResult.ValueForArgument(SlnRemoveParser.ProjectPathArgument)?.ToArray() ?? (IReadOnlyCollection<string>)Array.Empty<string>();
             if (_arguments.Count == 0)
             {
                 throw new GracefulException(CommonLocalizableStrings.SpecifyAtLeastOneProjectToRemove);
             }
 
-            _fileOrDirectory = parseResult.ValueForArgument<string>(SlnCommandParser.SlnArgument);
+            var slnFile = _arguments.FirstOrDefault(path => path.EndsWith(".sln"));
+            if (slnFile != null)
+            {
+                var projectArgs = string.Join(" ", _arguments.Where(path => !path.EndsWith(".sln")));
+                throw new GracefulException(new string[]
+                {
+                    string.Format(CommonLocalizableStrings.SolutionArgumentMisplaced, slnFile),
+                    CommonLocalizableStrings.DidYouMean,
```

Comment:
@KathleenDollard what do you think of the user experience here? The 'did you mean...' message feels odd to me, I think if anything we should provide an error message educating the user on the difference between a sln and proj as well as print the help here.

Reply:
I personally just find it faster to resolve problems when a tool can figure out and tell me what it expected instead of needing to spend time to figure out what it wanted but I can just remove the "Did you mean?" part if that's not what is wanted.
If you think its worth talking about it can be split out into an issue and discussed there instead maybe?

The help usage gets printed on stdout as part of `GracefulException` getting thrown.